pkg/controller/job/job_controller_test.go

```diff
@@ -492,21 +492,23 @@ func TestControllerSyncJob(t *testing.T) {
 			completions:        5,
 			backoffLimit:       6,
 			activePods:         3,
+			readyPods:          3,
 			expectedDeletions:  1,
 			expectedActive:     2,
 			expectedPodPatches: 1,
-			expectedReady:      ptr.To[int32](0),
+			expectedReady:      ptr.To[int32](2),
```

Comment:
Interesting, can you check why the number went up, since we now subtract the pods? 
I'm wondering if by putting the `ready` field in jobCtx we also fix another scenario.

Reply:
This is for the `too many active pods` test case.
```
"too many active pods": {
	parallelism:        2,
	completions:        5,
	backoffLimit:       6,
	activePods:         3,
	readyPods:          3,
	expectedDeletions:  1,
	expectedActive:     2,
	expectedPodPatches: 1,
	expectedReady:      ptr.To[int32](2),
}
```
 Active & ready pods are greather than parallelism, so the 1 extra pod gets deleted and expected ready should be 2.

Almost all test cases configured only active pods count, without setting ready count, so the test doesn't add the Ready condition.

Reply:
Thanks, so the current numbers make sense to me - the number of ready pods reduced from 3 -> 2 due to the deletion. However, I'm wondering why it was 0 on master.

Reply:
Ah, I got it, this is because you now add `readyPods` to the pre-existing tests. 

After the second thought I think I prefer to keep the pre-existing tests as they are, but add dedicated test cases as you did for "count ready pods when job fails". I think 3 new test cases per scenario: job failing, job suspended, and too many active pods.

Reply:
The tests lgtm after the recent update.